src/components/Channel/Channel.js

```diff
@@ -148,13 +155,15 @@ const ChannelInner = ({
       return;
     }
     lastRead.current = new Date();
-
-    logChatPromiseExecution(channel.markRead(), 'mark read');
-
+    if (doMarkReadRequest) {
+      doMarkReadRequest(channel);
+    } else {
+      logChatPromiseExecution(channel.markRead(), 'mark read');
+    }
```

Comment:
Hi @vishalnarkhede. We can define default `doMarkReadRequest` as bellow:
```tsx
function defaultMarkReadRequest(channel) {
  logChatPromiseExecution(channel.markRead(), 'mark read');
}
```

And this code will more clean, no unnecessary if-else:
```tsx
    doMarkReadRequest(channel)
```